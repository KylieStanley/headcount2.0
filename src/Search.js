import React, { Component } from 'react';
import './Search.css'


class Search extends Component {
  constructor() {
    super()
    this.state = {
      district: ''
    }
  }

  handleSearch = (e) => {
    this.setState({
      district: e.target.value
      }, () => this.props.matchCards(this.state.district)
    )
  }

  render() {
    return (
      <div className="search">
        <p>Search by District:</p>
        <input type="text" value={this.state.district} onChange={this.handleSearch}></input>
      </div>
    );
  }
}

export default Search;
