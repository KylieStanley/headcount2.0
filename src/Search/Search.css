@import "//netdna.bootstrapcdn.com/font-awesome/3.0/css/font-awesome.css";

.search {
  align-items: center;
  display: -ms-flexbox;
  display: flex;
  justify-content: center;
  margin: 20px auto;
  width: 500px;
}

p {
  margin-right: 5px;
}

input {
  font-size: 1.5rem;
  font-weight: bold;
  border-radius: 10px;
  border: 1px solid #e6e6e6;
  box-shadow: inset 2px 2px 1px 0px #ababab;
  font-family: 'Open Sans', "FontAwesome", sans-serif;
  padding: 5px;
  text-align: center;
  width: 400px;
}
