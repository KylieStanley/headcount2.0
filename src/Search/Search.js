import React, { Component } from 'react';
import './Search.css'
import PropTypes from 'prop-types';


class Search extends Component {
  constructor() {
    super()
    this.state = {
      district: ''
    }
  }

  handleSearch = (e) => {
    this.setState({
      district: e.target.value
      }, () => this.props.matchCards(this.state.district)
    )
  }

  render() {
    return (
      <div className="search">
        <input type="text" placeholder="&#xf002; Search by District" value={this.state.district} onChange={this.handleSearch}></input>
      </div>
    );
  }
}

Search.propTypes = {
  matchCards: PropTypes.func.isRequired
}


export default Search;
