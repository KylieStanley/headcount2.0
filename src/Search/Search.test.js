import React from 'react';
import Data from '../data/kindergartners_in_full_day_program.js';
import Search from './Search.js';
import {shallow, mount} from 'enzyme';

describe('Search', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Search 
      data={[]} 
      matchCards={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  })

  it('simulates text change in input box',() => {
    const mockData = { target: {value: 'Colorado'}}
    const wrapper = mount(<Search 
      data={[]} 
      matchCards={jest.fn()}
    />);

    const input = wrapper.find('input');
    const handleSearchSpy = jest.spyOn(wrapper.instance(), 'handleSearch')

    wrapper.instance().forceUpdate();
    input.simulate('change', mockData)

    expect(handleSearchSpy).toBeCalled()
  })

})