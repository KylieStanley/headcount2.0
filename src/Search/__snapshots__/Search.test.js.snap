// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    data={Array []}
    matchCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Search by District:
</p>,
        <input
          onChange={[Function]}
          type="text"
          value=""
/>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search by District:",
        },
        "ref": null,
        "rendered": "Search by District:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Search by District:
</p>,
          <input
            onChange={[Function]}
            type="text"
            value=""
/>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search by District:",
          },
          "ref": null,
          "rendered": "Search by District:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
